src/linker/Linker/Tracer.cs

```diff
@@ -70,6 +70,9 @@ bool IsRecordingEnabled ()
 
 		public void AddDirectDependency (object target, in DependencyInfo reason, bool marked)
 		{
+			if (reason.Kind == DependencyKind.Unspecified)
```

Comment:
Could we instead let each dependency recorder decide what to do with `Unspecified` dependencies?

Reply:
Sure.  Done.